spec/s3.spec.js

```diff
@@ -0,0 +1,192 @@
+    });
+    describe("configure", () => {
+        beforeEach(() => {
+            AwsSdkMock.S3 = jasmine.createSpy("s3Mock");
+        });
+        it("passes in configuration options for all", () => {
+            s3.transit();
+            expect(s3.aws.config.region).toBe("us-east-1");
+            expect(AwsSdkMock.S3).toHaveBeenCalledWith({params: {
+                Bucket: null
+            }});
+        });
+        it("passes in configuration options for all", () => {
+            s3.configure({
+                region: "us-west-1",
+                bucket: "test-bucket"
+            });
+            s3.transit();
+            expect(s3.aws.config.region).toBe("us-west-1");
+            expect(AwsSdkMock.S3).toHaveBeenCalledWith({params: {
+                Bucket: "test-bucket"
+            }});
+        });
+    });
+    describe("fileDel", () => {
+        var transit;
+        
+        beforeEach(() => {
+            transit = s3.transit();
+            transit.deleteObjectAsync.andCallFake((params) => {
+                if (params.Key == "afile") {
+                    return promiseMock.resolve("deleted file");
+                }
+
+                return promiseMock.reject("error deleting file");
+            });
+        });
+        it("deletes a file", () => {
+            checkPromiseResults(s3.fileDel("afile"), true, "deleted file");
+        });
+    });
+    describe("fileGet", () => {
+        var transit;
+        
+        beforeEach(() => {
+            transit = s3.transit();
+            transit.getObjectAsync.andCallFake((params) => {
+                if (params.Key == "afile") {
+                    return promiseMock.resolve("got data");
+                }
+
+                return promiseMock.reject("error fetching file");
+            });
+        });
+        it("gets an object back", () => {
+            checkPromiseResults(s3.fileGet("afile"), true, "got data");
+        });
+    });
+    describe("fileList", () => {
+        var transit;
+
+        beforeEach(() => {
+            transit = s3.transit();
+            transit.listObjectsAsync.andCallFake((params) => {
+                if (! params.Prefix) {
+                    return promiseMock.resolve("got list from top level");
+                }
+                
+                if (params.Prefix && params.Prefix === "accounts") {
+                    return promiseMock.resolve("got list from accounts");
+                } else {
+                    return promiseMock.reject("error fetching list");
+                }
+            });
+        });
+        it("gets top level list", () => {
+            checkPromiseResults(s3.fileList(), true, "got list from top level");
+        });
+        it("passes in a prefix", () => {
+            checkPromiseResults(s3.fileList("accounts"), true, "got list from accounts");
+        });
+    });
+    describe("filePut", () => {
+        var transit, expectedOptions;
```

Comment:
Both of these variables should not be defined here.  `transit` is only used in `beforeEach()` and `expectedOptions` is a bit weird to have here too.

`beforeEach()` should set up tests, but not perform testing.  You're setting up expectations in `beforeEach()` and that is a sign of a poorly designed test or an overly complicated method of testing some functionality.